plugin/src/main/java/org/opensearch/ml/action/connector/SearchConnectorTransportAction.java

```diff
@@ -78,12 +81,38 @@ private void search(SearchRequest request, ActionListener<SearchResponse> action
                 excludes.toArray(new String[0])
             );
             request.source().fetchSource(rebuiltFetchSourceContext);
+
+            ActionListener<SearchResponse> doubleWrappedListener = ActionListener.wrap(wrappedListener::onResponse, e -> {
+                if (e instanceof IndexNotFoundException) {
```

Comment:
better to use `ExceptionHelp.unwrap` to unwrap the exception for multiple node case.

Reply:
just update to `unwrapCause`

Reply:
Sounds good, will do

Reply:
@Hailong-am done